python/lsst/sims/coordUtils/Astrometry.py

```diff
@@ -888,18 +888,22 @@ def findChipName(self, xPupil=None, yPupil=None, ra=None, dec=None,
                 raise RuntimeError("No camera defined.  Cannot retrieve detector name.")
 
         chipNames = []
-        for x, y in zip(xPupil, yPupil):
-            cp = camera.makeCameraPoint(afwGeom.Point2D(x, y), PUPIL)
-            detList = [dd for dd in camera.findDetectors(cp) if dd.getType()==SCIENCE]
-            if len(detList) > 1 and not self.allow_multiple_chips:
-                raise RuntimeError("This method does not know how to deal with cameras where points can be"+
-                                   " on multiple detectors.  Override CameraCoords.get_chipName to add this.")
-            if not detList:
+        detList = camera.findDetectorsArray(xPupil, yPupil, PUPIL)
+        for det in detList:
+            if len(det)==0:
                 chipNames.append(None)
             else:
-                chipNames.append(detList[0].getName())
-
-        return numpy.asarray(chipNames)
+                names = [dd.getName() for dd in det if dd.getType()==SCIENCE]
+                if len(names)>1 and not self.allow_multiple_chips:
+                    raise RuntimeError("This method does not know how to deal with cameras " +
+                                       "where points can be on multiple detectors.  " +
+                                       "Override CameraCoords.get_chipName to add this.")
+                elif len(names)==0:
+                    chipNames.append(None)
+                else:
+                    chipNames.append(names[0])
+
+        return numpy.array(chipNames)
```

Comment:
Why do you cast this list into a numpy array? Why is this useful for a list of strings? It means you 
have to copy the data, and it's not clear what happens when a name is None.

If you insist on doing this then please document the dtype of the returned array, and what happens when a name is None. Also please consider returning a masked array, where None values are masked out. (But better to just return the list if you possibly can).

Right now it's dtype="0" (based on experimentation) which seems little more use than a list. On the other hand, a masked array of strings is a headache because numpy str arrays need a max number of chars per field, which is a headache (you have to compute it or set it to a fixed value and possibly truncate or waste space).